.github/workflows/benchmarking-regression.yml

```diff
@@ -0,0 +1,72 @@
+name: Benchmarking
+
+on:
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  actions: write
+  pull-requests: write
+
+run-name: Benchmarking from ${{ github.ref }}
+
+jobs:
+  benchmarking:
+    name: Benchmarking
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest]
+        python: ["3.10"]
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Set up Python 3.x
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
+        id: cached-pip-wheels
+        uses: actions/cache@v4
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+        
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Install dependancies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --sync
+      
+      - name: Install pytest-benchmark
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry run pip install pytest-benchmark
+
+      - name: Benchmarking on main
+        run: poetry run pytest --benchmark-only --benchmark-save="benchmark"
+
+      - uses: actions/checkout@v4
+        with:
+          clean: false
+
+      - name: Install dependancies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: 
+          poetry install --sync
```

Comment:
This if introduces a bug because we don't reload the cache the cache-hit is against the main poetry.lock.

Dropping the if statement seems fine or alternatively reloading the cache.